### ISSUE TITLE
Focus loss when switching to draft

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51901]
### Description

When switching a Published or Scheduled post to Draft:
- The Switch to Draft button gets a `disabled` attribute while saving.
- This triggers an immediate focus loss, as the element that had focus is now not focusable. See https://github.com/WordPress/gutenberg/issues/15524
- A confirm dialog opens, focus is moved to the dialog.
- WHen pressing OK to save as a Draft, the dialog opens and the Switch to Draft is removed from the DOM.
- As such, the dialog can't restore focus on the control that opened it.
- A focus loss occurs.
- At this point, when pressing the Tab key, the tab sequence starts from scratch from the document root.

Animated GIF to illustrate:

![switch to draft focus loss](https://github.com/WordPress/gutenberg/assets/1682452/6c6ff521-b50c-4aa0-89bd-c08a1b2f1589)

Worth noting the focus loss occurs also in WordPress 6.2, However, the Switch to Draft button was placed in the editor top bar. While the focus loss still occurs, I'd say it wa less serious because focus landed not that far from where it was.

Instead, with the new placement of the Switch to Draft button in the settings panel, users are now forced to navigate almost the entire editor UI to go back where they were.

The expected behavior is that focus is returned to the UI control that opened the modal dialog. As such, the 'Switch to Draft' button should not be removed from the DOM. Ideally, it shoul dstay in the DOM, be still focusable, use an `aria-disabled` attribute and 'noop'.  See: ARIA Authoring Practices: [Focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols).

### Step-by-step reproduction instructions

- Edit a Published or Scheduled post.
- Use the keyboard and navigate with the Tab key to the 'Switch to Draft' button in the Settings panel.
- Press Enter: a confirm dialog opens.
- Tab to the OK button and press Enter.
- The dialog closes.
- Observe the 'Switch to Draft' button is removed from the DOM.
- Press the Tab key.
- Observe the tab sequence starts from scratch from the Site icon button at the beginning of the document.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
